### PR TITLE
Fix the card to expand seperatly

### LINKED ISSUE #101: expanding 1 column expands the others
The fact that each column isn't independant is not great
Same for delegations

image

### PR DESCRIPTION
**Pull request by wirednkod**
closes #101

---

Submission checklist:

#### Layout

- [x] Fix cards to expand seperatly

image
image

### CODE REVIEW

**Review by Tbaut — CHANGES_REQUESTED** [review]
see above

**Comment by wirednkod** [comment]
> Can you please apply it to delegations too?

How many cards can exist there? In the first line there are max 3. In the "My delegations"?
I cannot create data due to the "ExhaustResources" bug

**Comment by Tbaut** [comment]
For the exhaust resource, juste undelegate a couple tracks with pjs and delegate again with other convictions or others delegate. There can be many more cards, but I expect ppl to have 2 or 3 max. The vast majority will only have 1.

**Comment by wirednkod** [comment]
image
image

**Comment by Tbaut** [comment]
it's not working on mobile any more :cry: 
image

I'd like to keep this grid really, as this is what tailwind is supposed to allow you to do quickly. I'll check if we can have grid columns that are independants and still responsive, without having to use flex here.

**Comment by wirednkod** [comment]
Ah.. my bad. I know why. Will fix in the morning

**Comment by Tbaut** [comment]
This PR is now exploding, with new dependencies... and by adding new things.. well I'm adding new opinions :see_no_evil: 
Let's try to keep it focused, and discuss about things before you spend time on it.

As I wrote above:
> There can be many more cards, but I expect ppl to have 2 or 3 max. The vast majority will only have 1.

Meaning that the carousel is effectively useless for the vast majority! Can we remove it unless there are many (more than 3) delegates?

The spacing is off with the cards above. If not needed, the carousel should be invisible

image

**Review by Tbaut — COMMENTED** [review]
I feel bad pushing back on this.. but as nice as it may be for many delegates, it's useless for most cases, and just makes things complex, while we never discussed it :see_no_evil:

File: src/components/LocksCard.tsx [hunk 1]
```diff
@@ -218,7 +205,7 @@ export const LocksCard = () => {
         </>
       ) : (
         <>
-          <Card className="h-full border-2 p-2 px-4">
+          <Card className="h-max border-2 p-2 px-4 md:col-span-1 md:col-start-1">
```
Comment (Tbaut) [inline src/components/LocksCard.tsx:208]: There's something wrong with this one, it always sits on top of the others :thinking: 
image

**Review by Tbaut — COMMENTED** [review]
That's very close to perfect, thank you so much!

File: src/components/LocksCard.tsx (hunk 1 above)
Comment (wirednkod) [reply src/components/LocksCard.tsx:208]: Ah yeah.. I never get the unlockable tab so its hard to identify :) thnx for pointing it out

Comment (Tbaut) [reply src/components/LocksCard.tsx:208]: Let's show it at all time, I don't see why this guy would be hidden when the others aren't :shrug:

Comment (wirednkod) [reply src/components/LocksCard.tsx]: image

File: src/components/LocksCard.tsx [hunk 2]
```diff
@@ -129,96 +128,90 @@ export const LocksCard = () => {
...
+            {freeLocks.map((lock) => {
+              if (lock.type === LockType.Delegating) {
+                const { amount, trackId } = lock
+                return (
+                  <div key={trackId}>
+                    <ul>
+                      <li className="mb-2">
+                        <div className="capitalize">
+                          <span className="capitalize">
+                            <Badge>{trackList[trackId]}</Badge> /{trackId}
```
Comment (Tbaut) [inline src/components/LocksCard.tsx:168]: Conflict went wrong here, it should be
```suggestion
                            <TrackDisplay trackId={trackId} />
```

File: src/components/LocksCard.tsx [hunk 3]
```diff
@@ -29,9 +29,8 @@ import { TrackDisplay } from './TrackDisplay'
 export const LocksCard = () => {
   const [currentBlock, setCurrentBlock] = useState(0)
   const [expectedBlockTime, setExpectedBlockTime] = useState(0)
-  const { api } = useNetwork()
+  const { api, assetInfo, trackList } = useNetwork()
```
Comment (Tbaut) [inline src/components/LocksCard.tsx:32]: with TrackDisplay there's no need from the list any more
```suggestion
  const { api, assetInfo } = useNetwork()
```

File: src/components/LocksCard.tsx [hunk 4]
```diff
@@ -227,9 +220,9 @@ export const LocksCard = () => {
                 </PopoverTrigger>
                 <PopoverContent>
                   <p className="max-w-[15rem]">
-                    Funds unlockable after a certain time. These locks result
-                    either from a vote with conviction casted on a refrendum, or
-                    from undelegating.
+                    Funds unlock after a certain time. These locks result either
```
Comment (Tbaut) [inline src/components/LocksCard.tsx:223]: I'd like to keep the same wording as the card, and if the card title is wrong, then let's change it. But the fact that users need to do something to get things unlocked makes me think "unlock" is not right. I agree unlockable is not great though..
```suggestion
                    Funds unlockable after a certain time. These locks result either
```

**Review by Tbaut — APPROVED** [review]
Super nice, looks great, just found a couple nits

**Pull request merged**